src/Bicep.LangServer/Completions/ModulesMetadataProvider.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft;
+using Newtonsoft.Json;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Bicep.LanguageServer.Providers
+{
+    public record ModuleMetadata(string moduleName, List<string> tags);
+
+    /// <summary>
+    /// Provider to get modules metadata from this endpoint - https://live-data.bicep.azure.com/modulesMetadata
+    /// The above endpoint helps fetch module names and versions of modules available in this github repository - https://github.com/Azure/bicep-registry-modules
+    /// </summary>
+    public class ModulesMetadataProvider : IModulesMetadataProvider
```

Comment:
>IModulesMetadataProvider [](http://example.com/codeflow?start=43&length=24)

Should we rename this to `IPublicRegistryModuleMetadataProvider`?

Reply:
Sure, renamed here:
https://github.com/Azure/bicep/commit/db6295fe366595495a866856be4de83e8d58da8b

Thanks!